lib/Twig/Template.php

```diff
@@ -437,9 +437,22 @@ protected function displayWithErrorHandling(array $context, array $blocks = arra
         if (!array_key_exists($item, $context)) {
             if ($ignoreStrictCheck || !$this->env->isStrictVariables()) {
                 return;
-            }
+            } else {
+                $errorMessage = sprintf('Variable "%s" does not exist', $item);
+                $templateName = $this->getTemplateName();
+                $lineno = -1;
+
+                if ($log = $this->env->isStrictVariablesNoticeLog() || $show = $this->env->isStrictVariablesNoticeShow()) {
```

Comment:
Without parenthesis around the assignments, it does not do what you expect it to.